lark/visitors.py

```diff
@@ -9,9 +10,14 @@
 
 ###{standalone
 from inspect import getmembers, getmro
-
-_Return_T = TypeVar('_Return_T')
-_Return_V = TypeVar('_Return_V')
+import sys
+if sys.version_info >= (3, 8):
+    from typing import Protocol
+else:
+    from typing_extensions import Protocol
+
+_Return_T = TypeVar('_Return_T', covariant=True)
+_Return_V = TypeVar('_Return_V', covariant=True)
```

Comment:
Quick question: Why is it Leaf_U but Return_V? Do the U and V stand for something?

Reply:
The signature for the multiplication operator has a lot of moving pieces. It starts with a `ITransformer[_Leaf_T, Tree[_Leaf_U]]` instance (a transformer that converts a `Tree[_Leaf_T`] to `Tree[_Leaf_U]`). Then a `ITransformer[_Leaf_U, _Return_V]` instance (a transformer that converts a `Tree[_Leaf_U`] to `_Return_V`).  Which results in a `ITransformer[_Leaf_T, _Return_V]` (a transformer that converts a `Tree[_Leaf_T`] to `_Return_V`).

There isn't a technical reason why it needs to be named with `_Return_V` instead of `_Return_U`. However, I was thinking that using `_Return_V` would make it clearer that the final type isn't necessarily related to the intermediary `_Leaf_U`. I'm open to switching to `_Leaf_U` or putting some of my above description into a comment in the source code if you think the current state should be changed.